rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java

```diff
@@ -315,6 +366,287 @@ public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionCon
             return cu;
         }
 
+        private static Set<J.Annotation> findAllAnnotations(J.CompilationUnit cu) {
+            Set<J.Annotation> annotations = new HashSet<>();
+            cu.getClasses().stream()
+                    .map(cl -> cl.getLeadingAnnotations())
+                    .flatMap(List::stream)
+                    .forEach(annotations::add);
+            cu.getTypesInUse().getTypesInUse().stream().forEach(type -> {
+                if (type instanceof JavaType.Class) {
+                    JavaType.Class aClass = (JavaType.Class) type;
+                    if (aClass.getKind() == JavaType.FullyQualified.Kind.Annotation) {
+                        FindAnnotations.find(cu, aClass.getFullyQualifiedName()).forEach(annotations::add);
+                    }
+                }
+            });
+            return annotations;
+        }
```

Comment:
I'd have expected something like the following, instead of all the scan methods below; would that be possible here?
```java
Set<J.Annotation> annotations = new JavaIsoVisitor<Set<J.Annotation> >() {
    public J.Annotation visitAnnotations(J.Annotations ann, Set<J.Annotation> set) {
        set.add(ann);
        return ann;
    }
}.reduce(cu, new HashSet<>());
```